pages/index.tsx

```diff
@@ -2,25 +2,81 @@ import type { NextPage } from "next";
 import React, { useState, useEffect } from "react";
 import { EuiText, EuiFieldText, EuiFlexGroup, EuiFlexItem, EuiButton } from "@elastic/eui";
 import { useRouter } from 'next/router';
+import Link from 'next/link'
+import { database } from '../firebaseConfig';
+import { setDoc, doc, collection, getDocs, where, query, DocumentData } from "firebase/firestore";
+
+const roomsCollection = collection(database,'rooms');
 
 const Home: NextPage = () => {
   const router = useRouter();
-  const [newRoom, setNewRoom] = useState("");
+  const [newRoom, setNewRoom] = useState('');
   const [joinRoom, setJoinRoom] = useState('');
+  const [roomList, setRoomList] = useState<any[]>([]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getActiveRooms = () => {
+    let docs: React.SetStateAction<any[]> = [];
+    getDocs(roomsCollection)
+      .then((data) => {
+        data.forEach((doc) => {
+          let id = doc.id;
+          let rd = doc.data();
+          rd.id = id;
+          docs.push(rd);
+        })
+        setRoomList(docs)
+      })
+  };
+
+  const createNewRoom = () => {
+    const currentRoom = newRoom;
+    const docID = "randomID"+ currentRoom
+    setDoc(doc(database, "rooms", docID), {
+      room: currentRoom,
+      users: []
+    }).then(() => {
+      setNewRoom('');
+    }).then(() => {
+      router.push(`/room/${newRoom}?roomid=${docID}`);
+    })
+  }
 
   useEffect(() => {
     if(!newRoom) {
       setNewRoom(roomGen());
+      getActiveRooms();
     }
-  }, [newRoom]);
+  }, [getActiveRooms, newRoom]);
 
   const roomGen = () => {
-    function a() {
-      return String.fromCharCode(65+Math.floor(Math.random() * 26));
-    }
+    const a = () => String.fromCharCode(65+Math.floor(Math.random() * 26));
     return a().concat(a(),a(),a());
   }
 
+  const joinExistingRoom = () => {
+    let matches: DocumentData[] = []
+    getDocs(query(roomsCollection, where("room", "==", joinRoom)))
+      .then(function(querySnapshot) {
+        querySnapshot.forEach(function(doc) {
+          let id = doc.id;
+          let rd = doc.data();
+          rd.id = id;
+          matches.push(rd);
+        });
+    })
+    .catch(function(error) {
+        console.log("Error getting documents: ", error);
+    })
+    .then(() => {
+      if (matches.length > 0){
+        router.push(`/room/${joinRoom}/?roomid=${matches[0].id}`)
+      } else {
+        window.alert("room does not exist");
+      }
+    });
+  }
```

Comment:
This function determines if the user supplied room input is an active room. It queries FB looking for a document where `room: joinRoom`. If there's no match, the `matches` array will be empty and for now an alert pops up that says the room does not exist. If there is a match, the rooms array will include one object. We take the ID of that object and add it as a query param then send the user to the room page.